routes/api/users.js

```diff
@@ -0,0 +1,135 @@
+const express = require('express')
+const router = express.Router()
+const user = require("../../models/user");
+const UserSchema = require('../../services/schemas/userSchema');
+const joi = require('../../utils/joi/joi');
+const jwt = require('jsonwebtoken');
+const secret = process.env.SECRET;
+const {auth} = require('../../authorization/auth');
+
+router.get('/', auth, async (req, res, next) => {
+  try {
+    const { email } = req.user
+      res.status(200).json({ 
+      status: 200,
+      data: `Authorization was successful: ${email}`
+    });
+  } catch (error) {
+    console.log(error)
+  }
+})
+
+router.post('/users/signup', async (req, res, next) => {
+  try {
+    const body = req.body;
+    const { email, password } = req.body
+    const result = joi.schemaRegistration.validate(body);
+    const { error } = result; 
+    const ifUserAlreadyExist = await UserSchema.findOne({email})
+    if (ifUserAlreadyExist) {
```

Comment:
tutaj kilka rzeczy do zmiany.
1. Nie ma potrzeby robic `const body`, bo go pozniej uzywasz tylko raz (i mozesz na L26 smialo zrobic `validate(req.body)` (roznica jest minimalna).
2. to `const result = joi.schemaRegistration.validate(body);` moze stac sie tym `const { error } = joi.schemaRegistration.validate(body);`
3. `const ifUserAlreadyExist = await UserSchema.findOne({email})`. Tutaj nie dostajesz z `findOne` wartosci typu Bool. Dostajesz usera, wiec zmien `ifUserAlreadyExist` na `user` i pozniej zrob `if (user)`.
4. Gdybys otrzymywal wartosc typu Bool, to zamiast `ifUserAlreadyExist`, lepiej bedzie Ci uzyc `userExists` albo `doesUserExist` (na podobnej zasadzie, jak robisz `hasPets` albo `isLoggedIn`). Starasz sie stworzyc rodzaj pytania, na ktore da sie odpowiedziec tak lub nie (true lub false). To nie jest latwe, i ludzie ucza sie "dobrze" nazywac rzeczy przez lata swoich karier. Ale gdziesz musisz zaczac :)